crates/sui-framework/packages/move-stdlib/sources/macros.move

```diff
@@ -84,7 +84,7 @@ public macro fun num_to_string($x: _): String {
     buffer.to_string()
 }
 
-public macro fun range_do($start: _, $stop: _, $f: |_|) {
+public macro fun range_do($start: _, $stop: _, $f: |_| -> _) {
```

Comment:
Can we not do `$R: drop` for these macros? Or do we not care since they are helpers? 
Or do we not care since we don't annotate the other types?

Reply:
@tnowacki I suggest we do all integer ones in a follow up. They won't be making any bytecode changes so should be an easy land.

Reply:
Gonna be a relatively big change.

Reply:
Nvm. I'll add it here since I already started.